crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -1446,7 +1446,11 @@ impl<'src> Parser<'src> {
         let mut elts = vec![first_element];
 
         self.parse_comma_separated_list(RecoveryContextKind::ListElements, |parser| {
-            elts.push(parser.parse_named_expression_or_higher().expr);
+            elts.push(
+                parser
+                    .parse_star_expression_or_higher(AllowNamedExpression::Yes)
+                    .expr,
+            );
```

Comment:
I don't understand this. Both methods correspond to different grammar rule. `parse_named_expression_or_higher` is for `named_expression` which does not limit the precedence of a starred expression while `parse_star_expression_or_higher` is a either  `star_expression` or `star_named_expression` where the starred expression should have a minimum precedence of a bitwise or (`|`) operator.

Reply:
Ah okay, that was not clear to me from reviewing the PR. I assumed it was simply a rename (because you changed all call sites but what i understand from your comment is that this isn't the case and this PR instead changes the parse rule)

Reply:
Yes, it changes to use the correct grammar rule for list expressions.